presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -3190,6 +3192,45 @@ public void testPrunePartitionFailure()
         assertUpdate("DROP TABLE test_prune_failure");
     }
 
+    @Test
+    public void testTemporaryStagingDirectorySessionProperties()
+    {
+        String tableName = "test_temporary_staging_directory_session_properties";
+        assertUpdate(format("CREATE TABLE %s(i int)", tableName));
+
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "temporary_staging_directory_enabled", "false")
+                .build();
+
+        HiveInsertTableHandle hiveInsertTableHandle = getHiveInsertTableHandle(session, tableName);
+        assertEquals(hiveInsertTableHandle.getLocationHandle().getWritePath(), hiveInsertTableHandle.getLocationHandle().getTargetPath());
+
+        session = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "temporary_staging_directory_path", "/tmp/custom/temporary-${USER}")
```

Comment:
nit: Maybe explicitly put "temporary_staging_directory_enabled": "true"?

I am asking this because I got confused when looking at it first time, then I realized temporary_staging_directory_enabled is by default true.